x/ibc/applications/transfer/module.go

```diff
@@ -193,6 +194,15 @@ func (am AppModule) OnChanOpenInit(
 	counterparty channeltypes.Counterparty,
 	version string,
 ) error {
+	// NOTE: for escrow address security only 2^32 channels are allowed to be created
```

Comment:
I would propose to refactor a bit, and pull out a common validation function like this:

```go
func ValidateTransferChannelParams(
	ctx sdk.Context,
	order channeltypes.Order,
	keeper keeper.Keeper,
	portID string,
	channelID string,
	version string,
) error {
	// NOTE: for escrow address security only 2^32 channels are allowed to be created
	// Issue: https://github.com/cosmos/cosmos-sdk/issues/7737
	channelSequence, err := channeltypes.ParseChannelSequence(channelID)
	if err != nil {
		return err
	}
	if channelSequence > math.MaxUint32 {
		return sdkerrors.Wrapf(types.ErrMaxTransferChannels, "channel sequence %d is greater than max allowed transfer channels %d", channelSequence, math.MaxUint32)
	}
	if order != channeltypes.UNORDERED {
		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s ", channeltypes.UNORDERED, order)
	}

	// Require portID is the portID transfer module is bound to
	boundPort := keeper.GetPort(ctx)
	if boundPort != portID {
		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid port: %s, expected %s", portID, boundPort)
	}

	if version != types.Version {
		return sdkerrors.Wrapf(types.ErrInvalidVersion, "got %s, expected %s", version, types.Version)
	}
	return nil
}
```

and then replace the lines here
https://github.com/cosmos/cosmos-sdk/blob/c5ef4f90317c1778ba57b231c0ea3c3df115cc36/x/ibc/applications/transfer/module.go#L197-L218

and here
https://github.com/cosmos/cosmos-sdk/blob/c5ef4f90317c1778ba57b231c0ea3c3df115cc36/x/ibc/applications/transfer/module.go#L240-L261

with the call to this function:
```go
	if err := ValidateTransferChannelParams(ctx, order, am.keeper, portID, channelID, version); err != nil {
		return err
	}
```